### PR TITLE
Fix Imperative publish

### PR DESCRIPTION
**Pull request by t1m0thyj**
🚨 Before deleting the branch, update the branch used by https://wash.zowe.org:8443/job/zowe-cli-deploy-component/

* Remove "prepare" script from NPM package before publishing. This should be safe to remove, as the "prepare" script only runs during the development lifecycle. The change was necessary to prevent errors during `npm pack`, since the "prepare" script `husky install` fails when only prod (not dev) dependencies are installed.
* Use 2 spaces instead of 4 in package.json files, because that's the standard indentation used by `npm init`
* Trim PKG_NAME and PKG_TAG parameters. If PKG_TAG is not trimmed, specifying a version like " 5.0.0-next.XXX" with a leading space causes the version to get published as `@latest` 😨

### CODE REVIEW

**Review by zFernand0 — APPROVED** [review]
LGTM

File: scripts/deploy-zowe-cli-component.groovy [hunk 1]
```diff
@@ -35,9 +35,9 @@ def rmProt(String url) {
 def opts = []
 opts.push(
     parameters([
-        string(name: 'PKG_NAME', defaultValue: 'cli', description: 'Name of the package to be deployed<br/><strong>Note:</strong> the <code>@zowe</code> scope will be prepended'),
-        string(name: 'PKG_TAG', defaultValue: 'daily', description: 'Tag to be distributed from artifactory'),
-        string(name: 'RECIPIENTS_LIST', defaultValue: '', description: 'List of emails to recevie build resutls (Override)')
+        string(name: 'PKG_NAME', defaultValue: 'cli', description: 'Name of the package to be deployed<br/><strong>Note:</strong> the <code>@zowe</code> scope will be prepended', trim: true),
+        string(name: 'PKG_TAG', defaultValue: 'daily', description: 'Tag to be distributed from artifactory', trim: true),
+        string(name: 'RECIPIENTS_LIST', defaultValue: '', description: 'List of emails to receive build resutls (Override)')
```
Comment (awharn) [inline scripts/deploy-zowe-cli-component.groovy:40]:
```suggestion
        string(name: 'RECIPIENTS_LIST', defaultValue: '', description: 'List of emails to receive build results (Override)')
```

Might be nice to fix this 🙂

File: scripts/repackage_tar.sh [hunk 2]
```diff
@@ -26,8 +26,13 @@ tar xzf $tarfile -C temp
 cd temp
 cd package
 # Unholy one liner which replace registry and repository with blank strings. Should convert this to javascript file soonTM.
+# Also remove prepare script which may require dev dependencies like Husky - https://github.com/typicode/husky/issues/914
 # Takes in package.json, outputs package_new.json
-node -e "package = require('./package.json');package.publishConfig.registry='$registry';package.version='$newversion';require('fs').writeFile('package_new.json', JSON.stringify(package, null, 4), 'utf8')"
+node -e "package = require('./package.json');
+    package.publishConfig.registry='$registry';
```
Comment (awharn) [inline scripts/repackage_tar.sh:32]: Do we need a continuation character?

File: scripts/deploy-zowe-cli-component.groovy (hunk 1 above)
Comment (t1m0thyj) [reply scripts/deploy-zowe-cli-component.groovy]: Thanks, fixed

**Pull request merged**